src/provider.tsx

```diff
@@ -1,31 +1,64 @@
-'use client';
+"use client";
 
-import * as React from 'react';
-import { createClient } from '@workos-inc/authkit-js';
-import { Context } from './context';
-import { Client, CreateClientOptions } from './types';
-import { initialState } from './state';
+import * as React from "react";
+import {
+  AuthenticationResponse,
+  createClient,
+  getClaims,
+} from "@workos-inc/authkit-js";
+import { Context } from "./context";
+import { Client, CreateClientOptions } from "./types";
+import { initialState } from "./state";
 
 interface AuthKitProviderProps extends CreateClientOptions {
   clientId: string;
   children: React.ReactNode;
 }
 
 export function AuthKitProvider(props: AuthKitProviderProps) {
-  const { clientId, devMode, apiHostname, https, port, redirectUri, children, onRedirectCallback } = props;
+  const {
+    clientId,
+    devMode,
+    apiHostname,
+    https,
+    port,
+    redirectUri,
+    children,
+    onRedirectCallback,
+  } = props;
   const [client, setClient] = React.useState<Client>(NOOP_CLIENT);
   const [state, setState] = React.useState(initialState);
 
+  const onRefresh = React.useCallback(
+    ({ user, accessToken, organizationId }: AuthenticationResponse) => {
+      const { role = null, permissions = [] } = getClaims(accessToken);
```

Comment:
If `permissions` is `undefined`, does that mean anything?

Double-checking that we want to default to any empty array in that case, or whether we should similarly propagate a `null` value like we have here for `role`.

Reply:
good question. the way i imagine developers will work with this is stuff like
```
{('edit:posts' in permissions) && <EditButton />}
```

I don't _think_ there is ever a value in checking if permissions has a value first.

Reply:
Just a heads up that I don't think `in` works with arrays, but if you swap it out with `.includes`, the example still stands! And there you'd appreciate not needing to worry about checking if `permissions` has a value.

Reply:
oh yeah whoops!